url78117c6033_BrianMoths_LittleManManager_tgz-pJ8-littlemangame_LittleManGameJ8/src/littlemangame/tutorial/tutorialoffice/components/IndicatorArrow.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package littlemangame.tutorial.tutorialoffice.components;
+
+import Renderer.Drawable;
+import java.awt.Graphics;
+
+/**
+ *
+ * @author brian
+ */
+public class IndicatorArrow implements Drawable {
+
+    static private final int arrowheadSmallnesFactor = 3;
+    private int tailX;
+    private int tailY;
+    private int headX;
+    private int headY;
+    private int animationCounter = 0;
+
+    public IndicatorArrow(int tailX, int tailY, int headX, int headY) {
+        this.tailX = tailX;
+        this.tailY = tailY;
+        this.headX = headX;
+        this.headY = headY;
+    }
+
+    @Override
+    public void draw(Graphics graphics) {
+        animationCounter++;
+        if (animationCounter / 30 % 2 == 0) {
+            drawShaft(graphics);
+            drawFirstArrowhead(graphics);
+            drawSecondArrowhead(graphics);
+        }
+    }
+
+    private void drawShaft(Graphics graphics) {
+        graphics.drawLine(tailX, tailY, headX, headY);
+    }
+
+    private void drawFirstArrowhead(Graphics graphics) {
+        final int displacementX = (tailX - headX) / arrowheadSmallnesFactor;
+        final int displacementY = (tailY - headY) / arrowheadSmallnesFactor;
+
+        final int arrowHeadX = headX + displacementX - displacementY;
+        final int arrowHeadY = headY + displacementY + displacementX;
+
+        graphics.drawLine(arrowHeadX, arrowHeadY, headX, headY);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 46:52, 58:64.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.